searcher/searcher.go

```diff
@@ -367,6 +365,13 @@ func newSearcher(
 		SpecialFiles:    wd.SpecialFiles(),
 	}
 
+	var vcsDir string
+	if (wd.IsLocal()) {
```

Comment:
Right. I'm saying that Driver should not have `IsLocal`. Rather than having the `Searcher` have special knowledge of what `IsLocal` means, it would be better to just move the path creation into the driver. That does mean that this new method will have identical implementations for all the existing drivers, but that's also the case with `IsLocal`.